src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -704,6 +705,11 @@ export class DebugService implements IDebugService {
 			}
 			this.endInitializingState();
 			this.cancelTokens(session.getId());
+
+			if (this.configurationService.getValue<IDebugConfiguration>('debug').closeReadonlyTabsOnEnd) {
+				const editorsToClose = this.editorService.getEditors(EditorsOrder.SEQUENTIAL).filter(({ editor }) => editor.resource?.scheme === DEBUG_SCHEME && editor.isReadonly);
```

Comment:
We should make sure the source actually came from the debug session that ended as well to avoid closing the wrong things.

There's a `getEncodedDebugData` on the `Source` in debugSource.ts that you can use to extract this information from the editor.resource URI.